songpal/method.py

```diff
@@ -152,18 +152,60 @@ async def __call__(self, *args, **kwargs):
 
     @property
     def inputs(self) -> Dict[str, type]:
-        """Input parameters for this method."""
-        return self.signature.input
+        """Input parameters for this method version."""
+        return (
+            self.signatures[self._version].input
+            if self.signatures[self._version].input is not None
+            else ""
+        )
+
+    @property
+    def latest_supported_version(self) -> str:
+        """Latest version supported by this method."""
+        return (
+            sorted(self._supported_versions, reverse=True)[0]
+            if len(self._supported_versions) > 0
+            else None
+        )
 
     @property
     def outputs(self) -> Dict[str, type]:
-        """Output parameters for this method."""
-        return self.signature.output
+        """Output parameters for this method version."""
+        return (
+            self.signatures[self._version].output
+            if self.signatures[self._version].output is not None
+            else ""
+        )
+
+    @property
+    def signature(self) -> MethodSignature:
+        """Method version signature."""
+        return self.signatures[self._version]
+
+    @property
+    def supported_versions(self) -> List[str]:
+        """List of supported version numbers for this method."""
+        return self._supported_versions
+
+    @property
+    def version(self) -> str:
+        """Method signature version number."""
+        return self._version
+
+    def add_supported_version(self, version: str):
+        """Add a supported version number for this method."""
+        if version not in self._supported_versions:
+            self._supported_versions.append(version)
```

Comment:
`_supported_versions` should be a set, so this would be simply:

```suggestion
        self._supported_versions.add(version)
```